internal/multigitter/shared.go

```diff
@@ -70,3 +73,61 @@ func ParseConflictStrategy(str string) (ConflictStrategy, error) {
 		return ConflictStrategyReplace, nil
 	}
 }
+
+// createTempDir creates a temporary directory in the given directory.
+// If the given directory is an empty string, it will use the os.TempDir()
+func createTempDir(cloneDir string) (string, error) {
+	if cloneDir == "" {
+		cloneDir = os.TempDir()
+	}
+
+	absDir, err := makeAbsolutePath(cloneDir)
+	if err != nil {
+		return "", err
+	}
+
+	err = createDirectoryIfDoesntExist(absDir)
+	if err != nil {
+		return "", err
+	}
+
+	tmpDir, err := os.MkdirTemp(absDir, "multi-git-changer-")
+	if err != nil {
+		return "", err
+	}
+
+	return tmpDir, nil
+}
+
+func createDirectoryIfDoesntExist(directoryPath string) error {
+	// Check if the directory exists
+	if _, err := os.Stat(directoryPath); !os.IsNotExist(err) {
+		return nil
+	}
+
+	// Create the directory
+	err := os.MkdirAll(directoryPath, os.ModePerm)
```

Comment:
0777 is higher permissions than needed for this directory. Just read and write access for the current user should be enough.